src/newcharacter.h

```diff
@@ -4,6 +4,7 @@
 
 #include <string>
 
+#include "cursesport.h"
```

Comment:
```suggestion
#include "cursesdef.h"
#include "cursesport.h"
```

including cursesdef.h is needed to silence errors in error in clang++-17 (idk why it did build on clang++-17 but warns it in clangd)
```
Field has incomplete type 'catacurses::window' (fix available)clang(field_incomplete_or_sizeless)
cursesport.h(16, 7): Forward declaration of 'catacurses::window'
```

also forward declaration of `avatar` is needed to fix

```
Unknown type name 'avatar' (fix available)clang(unknown_typename)
```

```c++
class avatar;
```